openapi.yml

```diff
@@ -0,0 +1,506 @@
+            application/json:
+              schema:
+                type: string
+  /workflow_templates/{workflow}:
+    get:
+      summary: Retrieve workflow template for given workflow
+      responses:
+        '200':
+          description: Workflow datastream found
+          content:
+            application/json:
+              schema:
+                type: string
+      parameters:
+        - name: workflow
+          in: path
+          required: true
+          example: accessionWF
+          schema:
+            $ref: '#/components/schemas/Workflow'
+  /workflow_archive:
+    get:
+      summary: Retrieve archived workflows
+      responses:
+        '200':
+          description: Request succeeded
+          content:
+            application/xml:
+              schema:
+                type: string
+      parameters:
+        - name: workflow
+          in: query
+          required: false
+          example: accessionWF
+          schema:
+            $ref: '#/components/schemas/Workflow'
+  /workflow_queue:
+    get:
+      summary: Retrieve the workflow queue
+      responses:
+        '200':
+          description: Request succeeded
+          content:
+            application/xml:
+              schema:
+                type: string
+      parameters:
+        - name: completed
+          in: query
+          required: false
+          example: dor:etdSubmitWF:catalog-status
+          schema:
+            type: string
+        - name: waiting
+          in: query
+          required: false
+          example: dor:etdSubmitWF:catalog-status
+          schema:
+            type: string
+  /workflow_queue/lane_ids:
+    get:
+      summary: Retrieve the workflow queue based on the provided lane_id
+      responses:
+        '200':
+          description: Request succeeded
+          content:
+            application/xml:
+              schema:
+                type: string
+      parameters:
+        - name: step
+          in: query
+          required: true
+          example: dor:etdSubmitWF:catalog-status
+          schema:
+            type: string
+  /workflow_queue/all_queued:
+    get:
+      summary: Retrieve the workflow queue for all queues
+      responses:
+        '200':
+          description: Request succeeded
+          content:
+            application/xml:
+              schema:
+                type: string
+      parameters:
+        - name: limit
+          in: query
+          required: false
+          example: '10'
+          schema:
+            type: string
+components:
+  schemas:
+    Druid:
+      description: Digital Repository Unique Identifier (bare or prefixed)
+      type: string
+      pattern: '^(druid:)?[b-df-hjkmnp-tv-z]{2}[0-9]{3}[b-df-hjkmnp-tv-z]{2}[0-9]{4}$'
+      example: 'druid:bc123df4567'
+    Process:
+      description: Process name
+      type: string
+      enum: [accessioning-initiate, approve, check-marc, checksum-compute, complete-ingest, content-metadata,
```

Comment:
I'm on the fence as to whether we should list processes and workflows as enums in this API spec. Pro: more validation. Con: another place to store this info and update it? Maybe it's the right place? I dunno. @sul-dlss/infrastructure-team ?

Reply:
Personally, I really like the idea of this validation. But I can be swayed either way be team consensus :-)